__tests__/pet.test.js/pet.test.js

```diff
@@ -0,0 +1,221 @@
+      });
+});
+
+describe('checkUp', () => {
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 5;
+        pet.fitness = 3;
+        
+        expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 0;
+        pet.fitness = 4;
+        
+        expect(pet.checkUp()).toEqual('I feel great!');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 0;
+        pet.fitness = 2;
+        
+        expect(pet.checkUp()).toEqual('I need a walk');
+
+    });
+
+    it('returns how the pet is feeling', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 6;
+        pet.fitness = 5;
+        
+        expect(pet.checkUp()).toEqual('I am hungry');
+
+    });
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+  
+        pet.age = 30;
+  
+        expect(() => pet.checkUp()).toThrow('Your pet is no longer alive :(');
+      });
+
+
+});
+
+describe('isAlive', () => {
+    it('returns if the pet is alive', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 7;
+        pet.fitness = 8;
+        pet.age = 20;
+
+        expect(pet.isAlive).toEqual(true);
+    });
+
+    it('returns if the pet is alive', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 7;
+        pet.fitness = 8;
+        pet.age = 30;
+
+        expect(pet.isAlive).toEqual(false);
+    });
+
+    it('returns if the pet is alive', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 7;
+        pet.fitness = 0;
+        pet.age = 20;
+
+        expect(pet.isAlive).toEqual(false);
+    });
+
+    it('returns if the pet is alive', () => {
+        const pet = new Pet('fido');
+
+        pet.hunger = 10;
+        pet.fitness = 8;
+        pet.age = 20;
+
+        expect(pet.isAlive).toEqual(false);
+    });
+
+    
+});
+
+describe('adoptChild', () => {
+    
+    it('returns the child onto the parent pet', () => {
+        const parent = new Pet('Dave');
+        const child = new Pet('Amelia');
+
+        expect(parent.adoptChild(child)).toEqual('You have adopted Amelia!');
+        
+
+
+    })
+      
+});
+
+describe('haveBaby', () => {
+    it('adds a new child to the parent' , () => {
+        const parent = new Pet('Dave');
+        parent.haveBaby('Billy');
+        
+        expect(parent.children).toEqual(expect.arrayContaining([{"age": 0, "children": [], "fitness": 10, "hunger": 0, "name": "Billy"}]));
+
+        parent.haveBaby('Bob');
+
+        expect(parent.children).toEqual(expect.arrayContaining([{"age": 0, "children": [], "fitness": 10, "hunger": 0, "name": "Billy"}], [{"age": 0, "children": [], "fitness": 10, "hunger": 0, "name": "Bob"}]));
```

Comment:
As before, this could be made much easier to read:
```
        expect(parent.children).toEqual(
            expect.arrayContaining([
                {
                    age: 0,
                    children: [],
                    fitness: 10,
                    hunger: 0,
                    name: "Billy",
                },
                { age: 0, children: [], fitness: 10, hunger: 0, name: "Bob" },
            ])
        );
```
Also this is more accurate, as arrayContaining now receives a single array with both the expected objects in